### DISCUSSION TITLE
meta-ros: superflore branches question ?

### DISCUSSION THREAD

**Discussion by SylvainPastor** [bmwcarit/meta-ros#1244] [Q&A] [answered]
Hello, community!

Philosophical question: 

When a new branch is created via the `superflore` tool (example: `superflore/kirkstone/humble/2024-09-19`), there are yocto errors because the versions in the generated-recipes directory have changed and not in the recipes-bbappends directory.

Examples
```
ERROR: No recipes in default available for:
  /home/sylvain/work/yocto/kirkstone_ws/layers/meta-ros2-humble/recipes-bbappends/backward-ros/backward-ros_1.0.2-3.bbappend
```
What is the correct approach or procedure when deciding to update your meta-ros and change branches in order to avoid manually redoing the bbappend files?

When someone in the community creates a new branch `superflore/[YOCTO VERSION]/[ROS_DISTRO]/[DATE]` to ensure that the versions of the bbapend files follow?

Or would you like me to provide a patch to fix it, for example?

My goal is just to clarify the process **when using a community generated superflore branch**.

Thanks,
Sylvain

**Comment by SylvainPastor** [comment]
Can I help you with testing or anything else?

**Reply by robwoolley** [reply, marked as answer]
Sorry for the delayed response.  I haven't been getting notifications on Discussion entries.  I think I may have fixed that now.

I personally use the rename-bbappends.sh script with the name of the layer and the latest "sync" commit
`scripts/rename-bbappends.sh meta-ros2-humble 6c1f040d792b15073e4e13fcb72712303d685f6d`

However, I don't merge the community generated superflore branches.  Since I have automated it in GitHub Actions to take less than 20 minutes, it is easier for me to re-run superflore myself rather than do a code review on the generated recipes.

Ideally, I would love to fully automate the whole process by having a scheduled CI script check for upstream ROS syncs.  With the GitLab CI work I have been doing it could even kick of a test build of ros-image-core and present the results and report any errors it encounters.

Let me know if that answers your question.   We are having our biweekly meeting in about an hour if you'd like to join: https://docs.google.com/document/d/1LqUjcu6vdlqVJO62SreCyjzddNDZhfO2n-7qYghY_cQ/edit?tab=t.0#heading=h.3vrwbu9vio04
